Marlin/Marlin_main.cpp

```diff
@@ -3576,10 +3576,28 @@ inline void gcode_M115() {
  * M117: Set LCD Status Message
  */
 inline void gcode_M117() {
-  char* codepos = strchr_pointer + 5;
+  millis_t codenum = 0;
+  char* codepos;
+  if (code_seen('S')) {
```

Comment:
Might as well keep `P` for compatibility – just in case Uncle Stuart or Aunt Martha wants to use `P4125` instead of `S4.125`.